MaterialLib/MPL/PropertyType.h

```diff
@@ -0,0 +1,274 @@
+/**
+ * \file
+ * \author Norbert Grunwald
+ * \date   Sep 7, 2017
+ *
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include <array>
+#include <boost/algorithm/string/predicate.hpp>
+#include <memory>
+#include <string>
+#include "BaseLib/Error.h"
+
+namespace MaterialPropertyLib
+{
+class Property;
+}
+
+namespace MaterialPropertyLib
+{
+/// PropertyType is an enumerator list of all known properties of a substance.
+/// This includes all properties on all scales (i.e. component, phase, and
+/// medium scales). It is used as an index for the PropertyArray of the
+/// materials. If a necessary property is not in the list, simply add a new one
+/// in alphabetical order (of course, except for the last entry). Please note
+/// that any of these entries must also appear in below convert functions.
+enum PropertyType : int
+{
+    acentric_factor,
+    binary_interaction_coefficient,
+    biot_coefficient,
+    brooks_corey_exponent,
+    bulk_modulus,
+    critical_density,
+    critical_pressure,
+    critical_temperature,
+    compressibility,
+    density,
+    drhodT,
+    effective_stress,
+    entry_pressure,
+    fredlund_parameters,
+    heat_capacity,
+    longitudinal_dispersivity,
+    molar_mass,
+    mole_fraction,
+    molecular_diffusion,
+    name,
+    permeability,
+    phase_velocity,
+    porosity,
+    reference_density,
+    reference_temperature,
+    reference_pressure,
+    relative_permeability,
+    residual_gas_saturation,
+    residual_liquid_saturation,
+    saturation,
+    specific_heat_capacity,
+    thermal_conductivity,
+    thermal_expansivity,
+    transversal_dispersivity,
+    viscosity,
+    number_of_properties
+};
+
+/// This function converts a string (e.g. a string from the configuration-tree)
+/// into one of the entries of the PropertyType enumerator. To avoid confusion,
+/// I suggest that the syntax of the properties in the input-files (i.e. the
+/// strings) have to be identical to the syntax of the entries in the
+/// enumerator.
+inline PropertyType convertStringToProperty(std::string const& inString)
```

Comment:
If we want to the elements of  PropertyType, we have to to change strings in  `convertStringToProperty` and `property_enum_to_string` as well.

It might be good to 
1) create a enum to std::string convertor in BaseLib (see https://stackoverflow.com/questions/28828957/enum-to-string-in-modern-c11-c14-c17-and-future-c20).
2) use the  enum to std::string convertor to initialize `std::array<std::string, PropertyType::number_of_properties>
    property_enum_to_string`
3) create a std::map<std::string, PropertyType> by using of std::string convertor to replace `convertStringToProperty` 

With that,   we only need to change the elements of PropertyType if such changed are needed.

The current implementation is also OK because  there might be less change in `PropertyType`.

Reply:
1) is not good. One still has to use the enum name to get its string type. It seems that each name of enum element must  appears at least two times in the source code. Therefore, please go forward with the present implementation.